### PR TITLE
Adapted to new fill

### PR DESCRIPTION
**Pull request by davidpiuva**

### CODE REVIEW

File: source/draw/Color.ooc [hunk 1]
```diff
@@ -133,13 +133,14 @@ ColorRgba: cover {
...
 	black ::= static This new(0, 0, 0, 255)
+	transparent ::= static This new(0, 0, 0, 0)
 	white ::= static This new(255, 255, 255, 255)
 	red ::= static This new(255, 0, 0, 255)
 	green ::= static This new(0, 255, 0, 255)
 	blue ::= static This new(0, 0, 255, 255)
 
 	init: func@ (=r, =g, =b, =a)
-	init: func@ ~default { this init(0, 0, 0, 0) }
+	init: func@ ~default { this init(0, 0, 0, 255) }
```
Comment (davidpiuva) [inline source/draw/Color.ooc]: Having transparent default RGBA made no sense when default RGB, YUV and YUVA is opaque black. The `transparent` color was added for backward compatibility.

File: source/plot/ScatterPlotData2D.ooc [hunk 2]
```diff
@@ -29,12 +29,12 @@ ScatterPlotData2D: class extends PlotData2D {
 	init: func ~default {
 		super()
 	}
-	init: func ~dataSeries (dataSeries: VectorList<FloatPoint2D>, label := "", colorRgba := ColorRgba new(), size := 5.f, shape := Shape Circle) {
+	init: func ~dataSeries (dataSeries: VectorList<FloatPoint2D>, label := "", colorRgba := ColorRgba transparent, size := 5.f, shape := Shape Circle) {
```
Comment (davidpiuva) [inline source/plot/ScatterPlotData2D.ooc]: This might not be what we wanted but that is how the tests are written now. Changes in behaviour should go in separate PR.

**Comment by davidpiuva** [comment]
@emilwestergren Peer review?

File: source/plot/LinePlotData2D.ooc [hunk 3]
```diff
@@ -24,14 +24,14 @@ LinePlotData2D: class extends PlotData2D {
 		super()
 		this lineStyle = lineStyle
 	}
-	init: func ~dataSeries (dataSeries: VectorList<FloatPoint2D>, label := "", colorRgba := ColorRgba new(), lineStyle := LineStyle Solid) {
+	init: func ~dataSeries (dataSeries: VectorList<FloatPoint2D>, label := "", colorRgba := ColorRgba transparent, lineStyle := LineStyle Solid) {
```
Comment (emilwestergren) [inline source/plot/LinePlotData2D.ooc]: You don't need to change this to `transparent`.

Comment (davidpiuva) [inline source/plot/LinePlotData2D.ooc]: The test failed using black. Am I allowed to change the expected result?

Comment (emilwestergren) [inline source/plot/LinePlotData2D.ooc]: Yes

**Comment by marcusnaslund** [comment]
Play with GPU fails.

File: source/plot/LinePlotData2D.ooc (hunk 3 above)
Comment (davidpiuva) [inline source/plot/LinePlotData2D.ooc]: By changing the none color to black, I did not have to change the output file since the color never made it to the output file.

File: source/draw/Color.ooc [hunk 4]
```diff
@@ -83,6 +83,7 @@ ColorYuva: cover {
 	y, u, v, a: Byte
 	normalized ::= FloatTuple4 new(this y as Float / 255, this u as Float / 255, this v as Float / 255, this a as Float / 255)
 	black ::= static This new(0, 128, 128, 255)
+	transparent ::= static This new(0, 128, 128, 0)
```
Comment (davidpiuva) [inline source/draw/Color.ooc]: If RGBA has `transparent` then YUVA should have it too for portability between the systems.

File: source/draw/Color.ooc [hunk 5]
```diff
@@ -133,13 +134,14 @@ ColorRgba: cover {
 	r, g, b, a: Byte
 	normalized ::= FloatTuple4 new(this r as Float / 255, this g as Float / 255, this b as Float / 255, this a as Float / 255)
 	black ::= static This new(0, 0, 0, 255)
+	transparent ::= static This new(0, 0, 0, 0)
```
Comment (davidpiuva) [inline source/draw/Color.ooc]: Some tests still use a transparent color for filling the background because it makes sense when representing a draw call that does not cover all pixels.

**Comment by davidpiuva** [comment]
Updated

**Comment by davidpiuva** [comment]
All tests passed locally.

**Comment by davidpiuva** [comment]
@emilwestergren Anything to add about the new changes?

**Comment by davidpiuva** [comment]
I ran PRInt again but something aborted the play test this time.

**Comment by marcusnaslund** [comment]
The play test?

**Comment by davidpiuva** [comment]
The last one that stopped.

**Comment by davidpiuva** [comment]
PRInt can not be accessed.

**Comment by marcusnaslund** [comment]
Already on it.

**Comment by marcusnaslund** [comment]
We're back online.

**Comment by marcusnaslund** [comment]
This looks good. Will just be testing it/them a bit manually before I merge. Should be upstream in 10 mins.

File: source/plot/SvgPlot.ooc [hunk 6]
```diff
@@ -105,7 +105,7 @@ SvgPlot: class {
 	}
 	setColor: func {
 		if (!datasets empty) {
-			noneColor := ColorRgba new(0, 0, 0, 0)
+			noneColor := ColorRgba black
```
Comment (marcusnaslund) [inline source/plot/SvgPlot.ooc]: `noneColor` went from invisible to pure black now?

Comment (davidpiuva) [inline source/plot/SvgPlot.ooc]: Yes. It makes no difference to the output.

Comment (marcusnaslund) [inline source/plot/SvgPlot.ooc]: What is `noneColor` then, if its value makes no difference?

Comment (davidpiuva) [inline source/plot/SvgPlot.ooc]: The color used for getting random colors.

Comment (marcusnaslund) [inline source/plot/SvgPlot.ooc]: I see now. So if you try to draw pure black, you will draw something else?

Comment (davidpiuva) [inline source/plot/SvgPlot.ooc]: No one is supposed to select color in the old broken version of the plotter.

Comment (marcusnaslund) [inline source/plot/SvgPlot.ooc]: Well, this *is* the plotter we have and will have for the forseeable future. Why was this value changed at all?

Comment (davidpiuva) [inline source/plot/SvgPlot.ooc]: Emil did not like writing transparent.

**Comment by marcusnaslund** [comment]
Alright then.
